contracts/non-fungible-tokens/andromeda-auction/src/contract.rs

```diff
@@ -437,6 +534,136 @@ fn execute_place_bid(
     ]))
 }
 
+fn execute_place_bid_cw20(
+    ctx: ExecuteContext,
+    token_id: String,
+    token_address: String,
+    amount_sent: Uint128,
+    asset_sent: AssetInfo,
+    // The user who sent the cw20
+    sender: &str,
+) -> Result<Response, ContractError> {
+    let ExecuteContext { deps, env, .. } = ctx;
+    let mut token_auction_state =
+        get_existing_token_auction_state(deps.storage, &token_id, &token_address)?;
+
+    let uses_whitelist = token_auction_state.whitelist;
+    if uses_whitelist {
+        ADOContract::default().is_permissioned_strict(
+            deps.storage,
+            env.clone(),
+            token_auction_state.auction_id,
+            sender,
+        )?;
+    };
+
+    ensure!(
+        !token_auction_state.is_cancelled,
+        ContractError::AuctionCancelled {}
+    );
+
+    ensure!(
+        token_auction_state.start_time.is_expired(&env.block),
+        ContractError::AuctionNotStarted {}
+    );
+    ensure!(
+        !token_auction_state.end_time.is_expired(&env.block),
+        ContractError::AuctionEnded {}
+    );
+
+    ensure!(
+        token_auction_state.owner != sender,
+        ContractError::TokenOwnerCannotBid {}
+    );
+
+    let sender_addr = deps.api.addr_validate(sender)?;
+
+    ensure!(
+        token_auction_state.high_bidder_addr != sender_addr,
+        ContractError::HighestBidderCannotOutBid {}
+    );
+
+    let auction_currency = token_auction_state.clone().coin_denom;
+    let valid_cw20_auction = ADOContract::default()
+        .is_permissioned_strict(
+            deps.storage,
+            env.clone(),
+            SEND_CW20_ACTION,
+            auction_currency.clone(),
+        )
+        .is_ok();
```

Comment:
Move this to StartSale, give error to prompt user to permission CW20 address. Only needs checked with `SEND_CW20_ACTION` is permissioned. 

This can be checked with `ADOContract::default().permissioned_actions.load(storage, SEND_CW20_ACTION)`.